### PR TITLE
[feature] book manager implementation and documentation

### LINKED ISSUE #11: [Suggestion] User Guide / Documentation
While playing around with the web app, it took me a while to figure out that the <kbd>PageUp</kbd> and <kbd>PageDown</kbd> keys were mapped.
After finding this github page and taking a look at the source code, I saw that the <kbd>A</kbd> and <kbd>D</kbd> keys were also mapped.

Maybe there should be a built-in user guide listing all the available keyboard shortcuts or a link to the (eventual) github documentation in the settings window.
Or did you plan on making the keyboard shortcuts fully customizable?

Something simple like this should do (could also include mouse controls, mobile gestures)
| Gesture/Shortcut Key | Description |
|-------------------------|--------------|
|<kbd>Escape</kbd>|Close Panels|
|<kbd>Space</kbd>|Toggle Auto-Scrolling|
|<kbd>A</kbd>,<kbd>D</kbd>|Adjust Auto-Scrolling Speed|
|<kbd>B</kbd>|Set Bookmark|
|<kbd>PageDown</kbd>,<kbd>PageUp</kbd>|Turn Pages|
|Swipe|...|

Another minor thing, but I also noticed a general lack of tooltips.

### PR DESCRIPTION
**Pull request by Renji-XD**
Implementation of #4.

Desktop Example:
https://imgur.com/a/8OfjpAo

Mobile Example:
https://imgur.com/a/TN9zB93

Additional Notes:

- Added some highlevel Page Description and Functionality Overview. Maybe enough to close #11 for the time being
- Added Logic to store the Progress String among bookmark Data. Thought its a nice little Detail for the Cover to show
- Added Logic to fallback to last item on navigation attempt instead of direct redirect to the file load screen
- Used hidden as ngIf took throttled / on mobile like 3 seconds alone to just open an empty modal (without any executed logic!) background.. with hidden everythings is ok and i just reset the state on close.. not enough angular experience to say why or how that can happen. with ngif..
- Implemented the Virtual Module with custom lazy Load of cover chunks for better general performance / for higher amount of books.. seems there are here and there some minor ux glitches in conjunction with the coverflow effect (like one cover which has to much rotation etc. until the next navigation) but nothing really big or discuptive (at least from what i saw) in my opinion

### CODE REVIEW

**Comment by ttu-ttu** [comment]
Some suggestions:
- Mousewheel to scroll through the list of books
- User can only open / highlight the book that's listed at the center, clicking on the side will just scroll to that specific book
- Clicking on the cover should open the book, not just on the text. Took me awhile to figure out how this works.
- Clicking on the book that's currently opened will just close the dialog
- Long click (hold) should highlight the book
- Trashcan icon shouldn't been shown in the top header, don't think this is used often enough

Think there's no need to make book highlighting feature explicit (by adding a checkbox), what do you think @wizo06 

In this context highlighting a book = select it in book manager page to prepare for deletion

**Comment by wizo06** [comment]
> Mousewheel to scroll through the list of books

The screenshots that @Renji-XD provided look like mousewheel scroll to me. If anything, it'd be nice if you can also hold&drag (desktop) or swipe (mobile) to scrol through the list of books.

---
> User can only open / highlight the book that's listed at the center, clicking on the side will just scroll to that specific book

I agree with this if we decide to go with the current "manager" view. However, i think changing the "manager" view to a grid would be even better. The grid view can solve the inconsistency of "if click on center book then select. if click on left/right book, then scroll to book". Not only that, i think the grid view is more user-friendly for large collection of imported books.

---
> Clicking on the cover should open the book, not just on the text. Took me awhile to figure out how this works.

I agree.

---
> Clicking on the book that's currently opened will just close the dialog

I agree.

---
> Long click (hold) should highlight the book

I don't think long click (desktop) is a common way of interacting with an interface. However, hold (mobile) *is* common.
The best compromise i can think of right now is to have a "select" button that allows the user to go into "select mode". In this "select mode":
- single click (desktop) or single tap (mobile) will select/deselect a book

The user can then click on the "delete" button after they've selected all the books they want to delete.

---
> Trashcan icon shouldn't been shown in the top header, don't think this is used often enough

I agree. Having the delete button outside and next to other buttons can also cause users to accidentally click on it, which can be frustrating. This is why some websites have the "log out" button far away from other commonly used buttons.

---
To summarize my suggestion:

- Move the "add book" (file-medical), "add folder" (folder-plus), and the "delete book" (trash) buttons inside the "manager view".
- Change the current "manager view" (horizontal scrolling of books, where left and right books are tilted) to grid style (imagine instagram).
- Inside the "manager view", a single click (desktop) or single tap (mobile) will open the book.
- Add a "select" button that allows the user to go into "select mode".
- Inside the "select mode", a single click (desktop) or single tap (mobile) will select/deselect a book.

Also, allow scrolling through the grid style manager view by mousewheel and hold&drag (for desktop) and by swiping (for mobile).

**Comment by Renji-XD** [comment]
hi,

- mousewheel is not added, swiping and grabbing is. Thought that is enough but no problem to add it

>  Clicking on the cover should open the book, not just on the text. Took me awhile to figure out how this works.

Had it at the beginning but quite often triggered the click handler by accident and then navigated without wanting it to a different book. Not sure if i am just a bad swiper or its my broken mouse : > Guess partially also solved by other comment to just navigate the centerd novel.. so i guess i can add it if you like

- trashcan / file / folder icon in bookmanager only. No hard feelings as i have my keybinds on desktop but for mobile user it would mean every time a couple of seconds longer to execute those actions than before (also on small mobile devices the headline already fills most of the width). Up to you ~

> Think there's no need to make book highlighting feature explicit (by adding a checkbox), [...]
In this context highlighting a book = select it in book manager page to prepare for deletion

Not really sure what you suggest here: you want the highlight replaced by checkboxes in order to indicate and execute the selection? I mean fine but in my opionion a plain checkbox would destroy the look&feel. Also on small Devices longer Titles nearly consuming all the space in landscape and would be maybe hard to hit among the title, while adding it on top of the cover would mean one line more to fill. For me a colored border is a less intrusive way of making clear what will be processed and what not.

> can also cause users to accidentally click on it

Both - bulk and single deletion have a confirmation dialog before executing it. Don't see a risk here - for me space is the only reason if you think it is a non often executed action (which i thought will basically the most used one ; ) ) and therefore should be removed from the topbar

> The best compromise i can think of right now is to have a "select" button 

What is the actual issue / concern we are trying to fix? For me adding a manual Select Button is nothing than one click more to do what i want. Also not sure what the difference in behavior is if I e. g. operate on the whole range of books - i would then be 100% in selection mode anyways. If its about potential Confusion of an User how to actually interaction with the books, maybe a info / question mark icon with a popover/popdown like "Single Tap on the current Book to navigate to it, Double Tap to select it for deletion" is already sufficient?

> However, i think changing the "manager" view to a grid would be even better. 

You mean something like a bookshelf in other reader applications? My Thoughts on it
- Especially on mobile titles already consume a lot of space which means you may run into vertical overflow when you have mutliple rows. So would be better to have a vertical direction in general instead i guess.
- Maybe even better to not display titles at all and just covers then? Otherwise you would artificially restrict the width in order to not create strange boxes 
- 3D Covers / Rotation then makes not really sense anymore
- On Desktop should still be height restricted.. otherwise i would feel "overflooded"
- May Lead to Longer Load Times again on Tablet / Mobile as you now load more Images at once i would expect
- Swiper supports multicolumns but not for the virtual Module.. which means if there should be something like virtual scrolling or infinite scrolling (which anyways does not exist in swiper) it either needs to be custom developed on top or again a different library needs to be added just for that which leaves the pure use case for swiper nearly for the swiping and column view.. sounds like a lot of required changes ; 0
- I would expect most of the Users will interact with max. 5 recent books at the same time (if not deleting them immediatly when done) and only rarely go over their whole library.. under that assumption not sure if it functional wise has any real benefit and is not only a personal preference in layout (which could be e. g. an later enhancement to switch between gallery and coverlist via buttons on top or whatever)
